src/components/Identification/index.js

```diff
@@ -85,48 +87,117 @@ export const Identification = props => {
   };
 
   /**
-   * Loads the user's profile info at start
+   * Loads the given user's profile info
    */
   useEffect(() => {
     async function loadUserProfile() {
       try {
-        const { def: defaultImage, pref: preferredImage } = await user.getImage();
-        const profile = await user.getProfileInfo();
-        setUserProfile(profile);
+        // Gets the given user's image. Depending on given user's person type and the currently
+        // signed-in user's person type, different images will be given
+        const { def: defaultImage, pref: preferredImage } =
+          props.profile.PersonType === 'fac'
+            ? /**
+               * The given user is Faculty
+               * If currently signed-in user is Faculty : Will receive default and preferred image
+               * If currently signed-in user is Non-Faculty : Will receive either default or preferred image
+               */
+              await user.getImage(props.profile.AD_Username)
+            : // Checks to see if the current page is the My Profile page
+            props.myProf
+            ? /**
+               * This case will occur only if the currently signed-in user is Non-Faculty and are
+               * on the My Profile page
+               */
+              await user.getImage()
+            : /**
+               * The given user is Non-Faculty
+               * If currently signed-in user is Faculty : Will receive default and preferred image
+               * If currently signed-in user is Non-Faculty : Will receive either default or preferred image
+               */
+              await user.getImage(props.profile.AD_Username);
+        setUserProfile(props.profile);
+        // Sets the given user's preferred image. If a default image is given but the preferred is undefined,
+        // then this could mean that the currently signed-in user is not allowed to see the preferred image or
+        // a preferred image wasn't set
```

Comment:
Fixes some bugs where the signed-in user would see the correct pictures on their My Profile page and their/others Public Profile pages